### ISSUE TITLE
Mismatch error

### ISSUE THREAD

**Issue by ghost** [anujdutt9/Handwritten-Digit-Recognition-using-Deep-Learning#15]
Hello, when I run the code knn.py I receive a mismatch error. Please see the attached. 
![MisMatch](https://user-images.githubusercontent.com/68824709/152821416-9c8fff56-14c5-438a-a8a6-02cedf0eaea4.PNG)

**Comment by anujdutt9** [comment]
Hi, it looks like you are getting an issue regarding the mismatch of the Magic Number in the dataset. The magic number for the dataset is available here: http://yann.lecun.com/exdb/mnist/
```
TRAINING SET LABEL FILE (train-labels-idx1-ubyte):
[offset] [type]          [value]          [description]
0000     32 bit integer  0x00000801(2049) magic number (MSB first)
0004     32 bit integer  60000            number of items
0008     unsigned byte   ??               label
0009     unsigned byte   ??               label
........
xxxx     unsigned byte   ??               label
The labels values are 0 to 9.

TRAINING SET IMAGE FILE (train-images-idx3-ubyte):
[offset] [type]          [value]          [description]
0000     32 bit integer  0x00000803(2051) magic number
0004     32 bit integer  60000            number of images
0008     32 bit integer  28               number of rows
0012     32 bit integer  28               number of columns
0016     unsigned byte   ??               pixel
0017     unsigned byte   ??               pixel
........
xxxx     unsigned byte   ??               pixel
Pixels are organized row-wise. Pixel values are 0 to 255. 0 means background (white), 255 means foreground (black).
```

So, can you please check that you have downloaded the following files from the link provided above?

[train-images-idx3-ubyte.gz](http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz):  training set images (9912422 bytes)
[train-labels-idx1-ubyte.gz](http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz):  training set labels (28881 bytes)
[t10k-images-idx3-ubyte.gz](http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz):   test set images (1648877 bytes)
[t10k-labels-idx1-ubyte.gz](http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz):   test set labels (4542 bytes)

If you still see the issue, then let me know.

**Comment by Sathyashree-srinivasan** [comment]
![Screenshot from 2023-01-09 20-39-51](https://user-images.githubusercontent.com/91139602/211341052-55242773-26cb-4f4b-92f2-7c5ba02f93c6.png)

Can you please tell me where it is wrong? :)

**Comment by anujdutt9** [comment]
Hi. Can you please extract the files from the attached zip file and see if that works. Let me know if you still have any issues.
Hope this helps.
[dataset.zip](https://github.com/anujdutt9/Handwritten-Digit-Recognition-using-Deep-Learning/files/10738095/dataset.zip)
